### PR TITLE
fix remarks Feature/groups

### PR DESCRIPTION
**Pull request by IvanKravchuk**

### CODE REVIEW

File: DTAdmin/Entities.swift [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    let groupId: String
+    let groupName: String
+    let facultyId: String
+    let specialityId: String
+    var facultyName: String?
+    var facultyDescription: String?
+    var specialityCode: String?
+    var specialityName: String?
+    
+    init(groupId: String,groupName: String,facultyId: String,specialityId: String) {
```
Comment (iostaf) [inline DTAdmin/Entities.swift]: Please put a space after commas

File: DTAdmin/Entities.swift [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        self.groupName = groupName
+        self.facultyId = facultyId
+        self.specialityId = specialityId
+        self.facultyName = nil
+        self.facultyDescription = nil
+        self.specialityCode = nil
+        self.specialityName = nil
+    }
+    
+    static func getGroupsFromJSON(json:[[String:String]]) -> [Group] {
```
Comment (iostaf) [inline DTAdmin/Entities.swift]: Put a space after colons

File: DTAdmin/Entities.swift [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    let facultyName: String
+    let facultyDescription: String
+    
+    init(facultyName: String, facultyId: String, facultyDescription: String) {
+        self.facultyId = facultyId
+        self.facultyName = facultyName
+        self.facultyDescription = facultyDescription
+    }
+    
+    static func getFacultiesFromJSON(json:[[String:String]]) -> [Faculty] {
```
Comment (iostaf) [inline DTAdmin/Entities.swift]: Put a space after colons

File: DTAdmin/Entities.swift [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+    let specialityCode: String
+    let specialityName: String
+    
+    init(specialityCode: String, specialityName: String, specialityId: String) {
+        self.specialityId = specialityId
+        self.specialityCode = specialityCode
+        self.specialityName = specialityName
+    }
+    
+    static func getSpecialitiesFromJSON(json:[[String:String]]) -> [Speciality] {
```
Comment (iostaf) [inline DTAdmin/Entities.swift]: Put a space after colons

File: DTAdmin/Entities.swift [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+            if let specialityId = faculty["speciality_id"],
+                let specialityCode = faculty["speciality_code"],
+                let specialityName = faculty["speciality_name"] {
+                specialities.append(Speciality(specialityCode: specialityCode, specialityName: specialityName, specialityId: specialityId))
+            }
+        }
+        return specialities
+    }
+}
+
```
Comment (iostaf) [inline DTAdmin/Entities.swift]: Remove extra new lines

File: DTAdmin/GroupViewController.swift [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+    @IBOutlet weak var groupTableView: UITableView!
+    
+    var commonDataForGroups = [Group]()
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        groupTableView.delegate = self
+        groupTableView.dataSource = self
+        
+        getCommonArrayForGroups(){(result:[Group]) in
```
Comment (iostaf) [inline DTAdmin/GroupViewController.swift]: Put a space before and after a `{` and after a colon.

File: DTAdmin/GroupViewController.swift [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+            DispatchQueue.main.async {
+                self.groupTableView.reloadData()
+            }
+        }
+    }
+    
+    func getCommonArrayForGroups (completion: @escaping ([Group]) -> ()) {
+        var groups = [Group]()
+        HTTPService.getAllData(entityName: "group") {
+            (groupJSON:[[String:String]],groupResponce) in
```
Comment (iostaf) [inline DTAdmin/GroupViewController.swift]: Put spaces after colons and commas

File: DTAdmin/GroupViewController.swift [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+            groups = Group.getGroupsFromJSON(json: groupJSON)
+            print(groups)
+            if groupResponce.statusCode == 200 {
+                HTTPService.getAllData(entityName: "faculty") {
+                    (facultyJSON:[[String:String]],facultyResponce) in
```
Comment (iostaf) [inline DTAdmin/GroupViewController.swift]: Put spaces after colons, commas

File: DTAdmin/GroupViewController.swift [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+                    let faculties = Faculty.getFacultiesFromJSON(json: facultyJSON)
+                    print(faculties)
+                    if facultyResponce.statusCode == 200 {
+                        HTTPService.getAllData(entityName: "speciality")  {
+                            (specialityJSON:[[String:String]],specialityResponce) in
```
Comment (iostaf) [inline DTAdmin/GroupViewController.swift]: Put spaces after colons, commas

File: DTAdmin/HTTPService.swift [hunk 10]
```diff
@@ -0,0 +1,215 @@
...
+                return
+            }
+            print("data = \(String(describing: data))")
+            do {
+                let json = try JSONSerialization.jsonObject(with: data, options: []) as? [String:String]
+                guard let response = response as? HTTPURLResponse, response.statusCode == 200 else {
+                    print("statusCode is not 200 while geting data with \(entityName) id \(id)")
+                    return
+                }
+                completion(json!, response)
```
Comment (iostaf) [inline DTAdmin/HTTPService.swift]: Do not use force unwrap Luke :)

File: DTAdmin/LoginViewController.swift [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+
+import UIKit
+
+class LoginViewController: UIViewController {
+    
+    @IBAction func loginButtoTapped(_ sender: Any) {
+        HTTPService.login(){(result:HTTPURLResponse) in
+            if result.statusCode == 200 {
+                let storyBoard: UIStoryboard = UIStoryboard(name: "GroupSB", bundle: nil)
+                let groupViewController = storyBoard.instantiateViewController(withIdentifier: "GroupVC") as! GroupViewController
```
Comment (iostaf) [inline DTAdmin/LoginViewController.swift]: Use GroupViewController instead fo GroupVC. Having the same identifier and custom class makes a life easier

File: DTAdmin/LoginViewController.swift [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+//
+
+import UIKit
+
+class LoginViewController: UIViewController {
+    
+    @IBAction func loginButtoTapped(_ sender: Any) {
+        HTTPService.login(){(result:HTTPURLResponse) in
+            if result.statusCode == 200 {
+                let storyBoard: UIStoryboard = UIStoryboard(name: "GroupSB", bundle: nil)
```
Comment (iostaf) [inline DTAdmin/LoginViewController.swift]: Use just Group instead fo GroupSB

**Review by iostaf — CHANGES_REQUESTED** [review]

File: DTAdmin/GroupViewController.swift (hunk 7 above)
Comment (dianapavaluc26) [reply DTAdmin/GroupViewController.swift]: Fixed in e4832b211. Assistant output persisted before response.created is now tracked and corrected after provider correlation, so the durable suppressHistoryReplay flag no longer depends on handler arrival order. Added a regression test that delivers transcript completion before response.created and verifies the persisted entry is repaired.

**Pull request closed without merging**